appengine/genetics/js/visualization.js

```diff
@@ -178,46 +261,50 @@ Genetics.Visualization.init = function() {
   };
   google.charts.load('current', {'packages': ['corechart']});
   google.charts.setOnLoadCallback(createCharts);
-};
 
-/**
- * Stop visualization.
- */
-Genetics.Visualization.stop = function() {
-  clearTimeout(Genetics.Visualization.pid);
+  Genetics.Visualization.display_ = document.getElementById('display');
 };
 
 /**
  * Clear chart data and set chart labels.
  * @private
  */
 Genetics.Visualization.resetChartData_ = function() {
-  Genetics.Visualization.populationChartWrapper_.setDataTable(
-      google.visualization.arrayToDataTable(
-          [[{label: 'Time', type: 'number'},
-            {label: Genetics.Mouse.Sex.MALE, type: 'number'},
-            {label: Genetics.Mouse.Sex.FEMALE, type: 'number'}]],
-          false));
-
-  // Create list of player labels and store order of players to use when
-  // population data updates.
-  Genetics.Visualization.playerOrder_.length = 0;
-  var playerLabels = [{label: 'Time', type: 'number'}];
-  for (var playerId in Genetics.Cage.players) {
-    if (Genetics.Cage.players.hasOwnProperty(playerId)) {
-      playerLabels.push({label: Genetics.Cage.players[playerId][0],
+  if (google.visualization) {
+    Genetics.Visualization.populationChartWrapper_.setDataTable(
+        google.visualization.arrayToDataTable(
+            [[{label: 'Time', type: 'number'},
+              {label: Genetics.Mouse.Sex.MALE, type: 'number'},
+              {label: Genetics.Mouse.Sex.FEMALE, type: 'number'}]],
+            false));
+
+    var playerLabels = [{label: 'Time', type: 'number'}];
+    for (var playerId = 0; playerId < Genetics.Cage.players.length;
+        playerId++) {
```

Comment:
for (var i = 0, player; player = Genetics.Cage.players[i]; i++) {
Then just use player.name below.